v2/cmd/controller/app/setup.go

```diff
@@ -314,6 +316,19 @@ func getDefaultAzureTokenCredential(cfg config.Values, setupLog logr.Logger) (az
 		return credential, nil
 	}
```

Comment:
You should add documentation (see https://github.com/Azure/azure-service-operator/pull/4577 for example) on this new option too.